napari/components/_viewer_key_bindings.py

```diff
@@ -59,15 +58,22 @@ def toggle_ndisplay(viewer: Viewer):
 @register_viewer_action(trans._("Toggle theme."))
 def toggle_theme(viewer: Viewer):
     """Toggle theme for viewer"""
-    settings = get_settings()
     themes = available_themes()
-    current_theme = settings.appearance.theme
+    current_theme = viewer.theme
+    # Check what the system theme is, to toggle properly
+    if current_theme == 'system':
+        current_theme = get_system_theme()
     idx = themes.index(current_theme)
     idx += 1
     if idx == len(themes):
         idx = 0
+    # Don't toggle to system, just among actual themes
+    if themes[idx] == 'system':
+        idx += 1
+        if idx == len(themes):
+            idx = 0
```

Comment:
```suggestion
    idx = themes.index(current_theme)
    idx += 1
    themes = themes + themes
    # Don't toggle to system, just among actual themes
    if themes[idx] == 'system':
        idx += 1
```

I'm not sure. It may be even less clear than modulo but maybe not. 
The core idea is to double the theme list, so there will be no need to perform a modulo operation or reset to zero.

Reply:
This is also really clever. Wow!

Reply:
For me, this method—with a comment on the `theme + theme` regarding the looping—seems like the best compromise between clarity and conciseness.
The modulo method is more clever though...

Reply:
My googling suggests that the modulo method is the accepted way of dealing with wrapping, so maybe it is the best way after all?

Reply:
Maybe. At this hour I'm not sure. maybe other people will have some opinion.

Reply:
Went with the modulo